#!/usr/bin/env python
PACKAGE = "trapezoidal_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("accSpeed", double_t, 0, "linear acceleration", 0.02, 0.0, 0.1)
gen.add("maxSpeed", double_t, 0, "max speed", 3.0, 0.0, 6.0)
gen.add("decSpeedGain", double_t, 0, "deceleration gain with speed", 0.2, 0.0, 3.0)
gen.add("direcLimit", double_t, 0, "direction limit (degree) for change direction", 30.0, 0.0, 180.0)
gen.add("speedLimit", double_t, 0, "speed limit (m/s) for change direction", 6.0, 0.0, 8.0)

gen.add("accRotation", double_t, 0, "rotation acceleration", 0.1, 0.0, 0.3)
gen.add("maxRotation", double_t, 0, "max rotation", 2.0, 0.0, 6.0)
gen.add("decRotationGain", double_t, 0, "deceleration gain with rotation", 0.2, 0.0, 3.0) 
gen.add("centrifugalLimit", double_t, 0, "centrifugalLimit for rotation control", 0.5, 0.0, 3.0)

exit(gen.generate(PACKAGE, "trapezoidal_control", "parameter"))
